Issue with CMake Configuration During Installation of mapmos

Hello,

I hope you're doing well. I am currently facing an issue with the CMake configuration while attempting to install the `mapmos` project. When running the `make editable` command, I encounter the following error:

```
CMake Error at /home/jonathan/MapMOS/build/_deps/sophus-src/CMakeLists.txt:1 (cmake_minimum_required):
    Compatibility with CMake < 3.5 has been removed from CMake.

    Update the VERSION argument <min> value.  Or, use the <min>...<max> syntax
    to tell CMake that the project requires at least <min> but has been updated
    to work with policies introduced by <max> or earlier.

    Or, add -DCMAKE_POLICY_VERSION_MINIMUM=3.5 to try configuring anyway.
```

I am using **CMake 4.0.0**, but the installation process suggests that the minimum required version of CMake is 3.5, and it is not compatible with older CMake configurations. The error message indicates that some settings need to be updated to support newer versions of CMake.

I have already attempted the following solutions in hopes of resolving the issue:

1. Updated CMake to version 4.0.0.
2. Used `-DCMAKE_POLICY_VERSION_MINIMUM=3.5` to force the minimum version setting.
3. Cleaned and rebuilt the project.

Unfortunately, the issue persists. If you have any insights or guidance on how to address this, I would greatly appreciate your help. Are there any recommended configuration changes or dependencies that I should adjust?

Thank you so much for your time and assistance, and I look forward to your advice!

Best regards,
Jonathan

--- a/src/mapmos/pybind/3rdparty/sophus/sophus.cmake
+++ b/src/mapmos/pybind/3rdparty/sophus/sophus.cmake
@@ -25,6 +25,5 @@ set(SOPHUS_USE_BASIC_LOGGING ON CACHE BOOL "Don't use fmt for Sophus libraru")
 set(BUILD_SOPHUS_TESTS OFF CACHE BOOL "Don't build Sophus tests")
 set(BUILD_SOPHUS_EXAMPLES OFF CACHE BOOL "Don't build Sophus Examples")
 
-# TODO: after https://github.com/strasdat/Sophus/pull/502 gets merged go back to mainstream
-FetchContent_Declare(sophus SYSTEM URL https://github.com/nachovizzo/Sophus/archive/refs/tags/1.22.11.tar.gz)
+FetchContent_Declare(sophus SYSTEM URL https://github.com/strasdat/Sophus/archive/refs/tags/1.24.6.tar.gz)
 FetchContent_MakeAvailable(sophus)

--- a/src/mapmos/pybind/3rdparty/tbb/tbb.cmake
+++ b/src/mapmos/pybind/3rdparty/tbb/tbb.cmake
@@ -20,7 +20,6 @@
 # LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 # OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 # SOFTWARE.
-# option(BUILD_SHARED_LIBS ON)
 option(BUILD_SHARED_LIBS OFF)
 option(TBBMALLOC_BUILD OFF)
 option(TBB_EXAMPLES OFF)
@@ -28,13 +27,17 @@ option(TBB_STRICT OFF)
 option(TBB_TEST OFF)
 
 include(FetchContent)
-FetchContent_Declare(tbb SYSTEM URL https://github.com/oneapi-src/oneTBB/archive/refs/tags/v2021.8.0.tar.gz)
+FetchContent_Declare(tbb URL https://github.com/uxlfoundation/oneTBB/archive/refs/tags/v2022.1.0.tar.gz)
+FetchContent_GetProperties(tbb)
 if(NOT tbb_POPULATED)
   FetchContent_Populate(tbb)
-  add_subdirectory(${tbb_SOURCE_DIR} ${tbb_BINARY_DIR} EXCLUDE_FROM_ALL)
-endif()
-
-if(${CMAKE_VERSION} VERSION_LESS 3.25)
-  get_target_property(tbb_include_dirs tbb INTERFACE_INCLUDE_DIRECTORIES)
-  set_target_properties(tbb PROPERTIES INTERFACE_SYSTEM_INCLUDE_DIRECTORIES "${tbb_include_dirs}")
+  if(${CMAKE_VERSION} GREATER_EQUAL 3.25)
+    add_subdirectory(${tbb_SOURCE_DIR} ${tbb_BINARY_DIR} SYSTEM EXCLUDE_FROM_ALL)
+  else()
+    # Emulate the SYSTEM flag introduced in CMake 3.25. Withouth this flag the compiler will
+    # consider this 3rdparty headers as source code and fail due the -Werror flag.
+    add_subdirectory(${tbb_SOURCE_DIR} ${tbb_BINARY_DIR} EXCLUDE_FROM_ALL)
+    get_target_property(tbb_include_dirs tbb INTERFACE_INCLUDE_DIRECTORIES)
+    set_target_properties(tbb PROPERTIES INTERFACE_SYSTEM_INCLUDE_DIRECTORIES "${tbb_include_dirs}")
+  endif()
 endif()

--- a/src/mapmos/pybind/3rdparty/tsl_robin/tsl_robin.cmake
+++ b/src/mapmos/pybind/3rdparty/tsl_robin/tsl_robin.cmake
@@ -21,5 +21,5 @@
 # OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 # SOFTWARE.
 include(FetchContent)
-FetchContent_Declare(tessil SYSTEM URL https://github.com/Tessil/robin-map/archive/refs/tags/v1.2.1.tar.gz)
+FetchContent_Declare(tessil SYSTEM URL https://github.com/Tessil/robin-map/archive/refs/tags/v1.4.0.tar.gz)
 FetchContent_MakeAvailable(tessil)

